### DISCUSSION TITLE
Can't cache documents?

### DISCUSSION THREAD

**Discussion by reggi** [freshframework/fresh#748] [Q&A]
Currently you can't cache a document because you can't change the status response using `Etag` / `If-None-Match`, curious if there's a reason this was only done for static content and not pages?

```ts
export const handler: Handlers = {
  async GET(req, ctx) {
    const resp = await ctx.render();
    
    const x = req.headers.get('If-None-Match')
    if (x === 'W/deadbeef') {
      return resp // need to send 304
    }
    
    resp.headers.set("ETag", "deadbeef");
    return resp;
  },
};
```

**Comment by lucacasonato** [comment]
I don't quite understand - why can you not set the `Etag` header on the response object? That should work.

**Reply by reggi** [reply]
From what I read I needed to also return a http status code of 304 after I confirm that the `If-None-Match` matches.

Regardless of this specific application I think it's odd we're not able to change the status code of a route arbitrarily, even with the new implementation of 404 it's still abstracted out.

**Reply by lucacasonato** [reply]
Then you should replace the line `return resp // need to send 304` with `return new Response("", { status: 304 });`.

Also you should do this before rendering, otherwise there is not going to be much benefit.

**Comment by reggi** [comment]
I believe the correct way to do this is:

```ts
export const handler: Handlers = {
  async GET(req, ctx) {    
    const x = req.headers.get('If-None-Match')
    // replace this with a unique string for each page to establish a cache criteria
    const id = 'deadbeef'
    if (x === `W/${id}`) {
       return new Response("", { status: 304 });
    }
    resp.headers.set("ETag", id);
    return await ctx.render();
  },
};
```
